Dashboard/Client/SessionManagement/ClientSessionManager.cs

```diff
@@ -217,16 +222,42 @@ public void OnDataReceived(string serializedData)
         /// </summary>
         public void RemoveClient()
         {
-            ClientToServerData clientToServerData = new("removeClient", _user.username, _user.userID);
-            string serializedData = _serializer.Serialize<ClientToServerData>(clientToServerData);
-            _communicator.Send(serializedData, moduleIdentifier);
+            SendDataToServer("removeClient", _user.username, _user.userID);
+        }
+
+        /// <summary>
+        /// A helper function that sends data from Client to the server side. The data consists
+        /// of the event type and the user who requested for that event.
+        /// </summary>
+        /// <param name="eventName"> The type of the event. </param>
+        /// <param name="username"> The username of the user who requested</param>
+        /// <param name="userID"> The user ID of the user, if the user has not yet beend created then the default value is -1. </param>
+        public void SendDataToServer(string eventName, string username, int userID = -1)
+        {
+            ClientToServerData clientToServerData;
+            lock(this)
+            {
+                clientToServerData = new(eventName, username, userID);
+                string serializedData = _serializer.Serialize<ClientToServerData>(clientToServerData);
+                _communicator.Send(serializedData, moduleIdentifier);
+            }
+
         }
 
+        /// <summary>
+        /// Used to set/change the Client side UserData Object for testing and deubgging purposes.
+        /// </summary>
+        /// <param name="userName"> The username of the user.</param>
+        /// <param name="userID"> The user ID of the user, the default value is -1. </param>
         public void SetUser(string userName, int userID = 1)
         {
             _user = new UserData(userName, userID);
         }
 
+        /// <summary>
+        /// Used to set/change the users list for testing and deubgging purposes.
+        /// </summary>
+        /// <param name="users">The list of UserData object that the sessionData will be asssignet to.</param>
         public void SetSession(List<UserData> users)

```

Comment:
Shouldn't this take a session object to set rather than users list?

Reply:
Yes it absolutely can, but this way saves me some lines of code, and this function is not to be used by any other module. Should I change it, or I can keep it like that?

Reply:
Can we rename it... to SetSessionUsers

Reply:
Sure.